File: homework04.js
function onErr(err) {
  console.log(err);
  return 1;
}

function main() {
  //initalize variables
  
  var height = 0,
    diameter = 0,
    volume = 0,
    surfaceArea = 0,
    unitMeasure = "";

  var splitNumbers = [];
  //introduction/explanation
  console.log("Input two the height, diameter, and unit for a cone, and I will output the Surface Area and Volume");
  
  //prompt for unit measurement
  //prompt for diameter and height
  //save diameter and height to their respective variables
  var prompt = require('prompt');

  var schema = {
    properties: {
      Values: {
        pattern: /(\d+)\s+(\d+)\s+(\w+)/,
        message: 'The format should be: height diameter unit measurement. Do not forget the space in between each value.',
        required: true
      }
    }
  };

  prompt.start();

  prompt.get(schema, function (err, result) {
    if (err) { return onErr(err); }

    console.log('  Values: ' + result.Values);
    splitNumbers = result.Values.split(" ");
    height = splitNumbers[0];
    console.log("\theight = " + height);
    diameter = splitNumbers[1];
    console.log("\tdiameter = " + diameter);
    unitMeasure = splitNumbers[2];
    console.log("\tunitMeasure = " + unitMeasure);
    
    //calculate answer
    var radius = diameter / 2;
    surfaceArea = Math.PI * radius * (radius + Math.sqrt(Math.pow(height, 2) + Math.pow(radius, 2)));
    volume = ((Math.PI * Math.pow(radius, 2)) * height) / 3;
    var sA = surfaceArea.toFixed(4);
    var v = volume.toFixed(4);
    
    //display answer with measurment
    console.log("A cone with a height of " + height + " " + unitMeasure + " and a diameter of " + diameter + " " + unitMeasure);
    console.log("will have a Surface Area of " + sA + " square " + unitMeasure);
    console.log("and a Volume of " + v + " cubic " + unitMeasure);
  });
  
  //All done!
}
 
// run the program
main();

File: homework04.cpp
/*******************************************************************************
********************************************************************************
**
**  Nathan A. Zobrist
**  Assignment #4
**  CSC 135, Section 1
**
********************************************************************************
**
**  This program is designed to calculate the volume and surface area of a
**    cone given its diameter and height.
**
**  input: height and diameter in numerical form, unit of measurement for cone
**  output: print the diameter and height of the cone as well as its surface
**    area and volume with appropriate unit of measurement labels.
**
********************************************************************************
*******************************************************************************/


#include <iostream>
#include <cmath>	// pow()
#include <iomanip>	// setprecision()
#include <string>	

using namespace std;

const float PI = 3.14159;

int main ()
{

	float diameter = 0, height = 0;		// diameter and height
	float volume = 0, surArea = 0;		// volume and surface area
	string units;// = NULL;			// units of measurement

	cout << setiosflags (ios::fixed) << setprecision (2);
 
	cout << "\nThis program is designed to calculate the volume and"
	     << "\n\t surface area of a cone given it's diameter and height.";

	cout << "\n\nPlease enter the diameter of the cone: ";
	cin >> diameter;

	cout << "\nPlease enter the height of the cone: ";
	cin >> height;

	cout << "\nWhat is the unit of measurement for the cone? ";
	cin >> units;

	float r = .5 * diameter,						// r = raduis
	      s = sqrt(pow(r,2) + pow(height,2)),		// s = slant height
	      C = 2 * PI * r,							// C = circumference
	      A = PI * pow(r,2);						// A = area of base
	  // these formulas were given as part of the problem


	surArea = A + .5 * C * s;		// calculate surface area
	volume = 1.0/3.0 * A * height;	// calculate volume

	cout << "\n\nFor a cone of height " << height << ' ' << units
	     << " and diameter of " << diameter << ' ' << units  << endl
	     << " surface area is: " << surArea << " square " << units  << endl
	     << " and volume is: " << volume << " cubic " << units << '.'  << endl;

	return 0;
}
